packages/tc-schema-sdk/data-accessors/relationship-type.js

```diff
@@ -0,0 +1,141 @@
+const primitiveTypesMap = require('../lib/primitive-types-map');
+const metaProperties = require('../lib/meta-properties');
+const TreecreeperUserError = require('../lib/biz-ops-error');
+
+const BIZ_OPS = 'biz-ops';
+
+const transformRichRelationship = (
+	rootType,
+	property,
+	{ name, from, to, relationship, properties },
+) => {
+	const { direction } = property;
+	let relationshipTo;
+	let hasMany;
+	let dest;
+	if (from.type === to.type && !direction) {
+		throw new TreecreeperUserError(
```

Comment:
shouldn't this check be in the tc-schema-validator library? In general, the sdk does not validate that the schema rawData is of the expected structure. If you need to throw here in order to prevent the code later on executing successfully (but making bad decisions) throw a plain error, not a TreecreeperUserError (which are reserved for when a user passes bad data to the api, not when the maintainer of the application has misconfigured something)